src/main/java/org/junitpioneer/jupiter/AbstractEntryBasedExtension.java

```diff
@@ -36,20 +37,28 @@
 import org.junit.jupiter.api.extension.ExtensionContext.Namespace;
 import org.junit.jupiter.api.extension.ExtensionContext.Store;
 import org.junit.platform.commons.support.AnnotationSupport;
+import org.junitpioneer.internal.PioneerAnnotationUtils;
 import org.junitpioneer.internal.PioneerUtils;
 
 /**
  * An abstract base class for entry-based extensions, where entries (key-value
- * pairs) can be cleared or set.
+ * pairs) can be cleared, set, or restored.
  *
  * @param <K> The entry key type.
  * @param <V> The entry value type.
  * @param <C> The clear annotation type.
  * @param <S> The set annotation type.
+ * @param <R> The restore annotation type.
  */
-abstract class AbstractEntryBasedExtension<K, V, C extends Annotation, S extends Annotation>
+abstract class AbstractEntryBasedExtension<K, V, C extends Annotation, S extends Annotation, R extends Annotation>
 		implements BeforeEachCallback, AfterEachCallback, BeforeAllCallback, AfterAllCallback {
 
+	/** Key to indicate storage is for an incremental backup object */
+	private static final String INCREMENTAL_KEY = "inc";
+
+	/** Key to indicate storage is for a complete backup object */
+	private static final String COMPLETE_KEY = "full";
```

Comment:
I added the class name to the `getStoreKey()` method since this class is abstract and used for SysProps and EnvVars.  Let me know if you would like a different solution.